flow/environment.py

```diff
@@ -346,14 +358,8 @@ def get_prefix(cls, operation, parallel=False, mpi_prefix=None, cmd_prefix=None)
     def _get_default_directives(cls):
         return _Directives(
             [
-                _NP,
-                _NGPU,
-                _NRANKS,
-                _OMP_NUM_THREADS,
-                _GET_EXECUTABLE(),
-                _WALLTIME,
-                _PROCESSOR_FRACTION,
-                _MEMORY,
+                directive() if callable(directive) else directive
```

Comment:
We can't perform this because all directives are callable :sweat_smile:. We use a callable to return another callable for the executable directive.

Reply:
Fixed.